services/http/api/lib/models/initializers.ts

```diff
@@ -0,0 +1,202 @@
+import { ModelFactory, Neogma } from "neogma";
+import {
+    user,
+    measurement,
+    resource,
+    identity,
+    date
+} from '../models';
+
+export function initDateModel(db: Neogma): date.DateModel {
+    return ModelFactory<date.DateProperties, date.DateRelatedNodes>(
+        {
+            label: 'Date',
+            primaryKeyField: 'id',
+            schema: {
+                id: {
+                    type: 'string',
+                    required: true
+                },
+                day: {
+                    type: 'number',
+                    required: true
+                },
+                month: {
+                    type: 'number',
+                    required: true
+                },
+                year: {
+                    type: 'number',
+                    required: true
```

Comment:
I'm noticing that these are being defined in two places - the model itself and then again in this initializers file. Would it be possible to centralize the model definition in one place and use it across everything instead? Do you see any downsides of doing so? Alternatively (if that doesn't work or isn't a clean solution) we can define it partially in the model, but then reuse those portions of the definition in here rather than redefining them here, maybe?

Reply:
They aren't redefined, actually, it's just a way for Neogma to enforce those types that were inputted with the `IdentityProperties` type. I've tried using an exported JS object, but TypeScript complained about it not being the right type.
```.ts
export const IdentitySchema = {
    type: {
        type: 'string',
        required: true
    },
    name: {
        type: 'string',
        required: true
    },
    check: {
        type: 'string',
        required: true
    },
}
```

```.ts
/*
Type '{ type: { type: string; required: boolean; }; name: { type: string; required: boolean; }; check: { type: string; required: boolean; }; }' 
is not assignable to type '{ type: ISchema<IdentityProperties> | JSONSchema<IdentityProperties>; name: ISchema<IdentityProperties> | JSONSchema<...>; check: ISchema<...> | JSONSchema<...>; }'.
*/
```

Types flow like this: `Our Custom Property Type -> Neogma Model Factory -> Model Schema`.
If one were to change the schema in the `IdentityProperties` type 
```.ts
export type IdentityProperties = {
    type: string,
    name: string,
    check: string
}
```
Neogma would ask for a change in the schema to mirror the change in the type.

Reply:
Update: 
In #13, I've made it so that the initializers live with the model definition.